File: proyecto-veterinaria-laravel/app/Models/Mascota.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory; // Assuming HasFactory is needed
use Illuminate\Database\Eloquent\Concerns\HasUuids; // Assuming HasUuids is needed

class Mascota extends Model
{
    use HasFactory, HasUuids;

    protected $fillable = [
        'usuario_id',
        'nombre',
        'tipo',
        'raza',
        'edad',
        'peso',
        'genero',
        'notas_medicas',
    ];

    public function dueno()
    {
        return $this->belongsTo(Usuario::class, 'usuario_id');
    }

    public function citas()
    {
        return $this->hasMany(Cita::class);
    }

    public function historial()
    {
        return $this->hasMany(HistorialMedico::class);
    }
}

File: maskotas/src/main/java/com/maskotas/model/Mascota.java
package com.maskotas.model;

import jakarta.persistence.*;
import lombok.Data;
import java.util.List;

@Entity
@Table(name = "mascotas")
@Data
public class Mascota {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nombre;
    private String tipo;
    private String raza;
    private Integer edad;
    private Double peso;
    private String genero;
    private String notasMedicas;

    @ManyToOne
    @JoinColumn(name = "usuario_id")
    private Usuario dueno;

    @OneToMany(mappedBy = "mascota")
    private List<Cita> citas;

    @OneToMany(mappedBy = "mascota")
    private List<HistorialMedico> historial;
}
